.github/workflows/test.yml

```diff
@@ -0,0 +1,48 @@
+name: Tests
+
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
+
+jobs:
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: ['1.21.x']
```

Comment:
The most recent version of Go is 1.23.3, so if you don't need to guarantee compilation against legacy versions it looks a proper way to always use the most recent versions and update as soon as possible. The same is related to dependencies since they could fix some security issues or bugs on new releases.